force-app/main/default/classes/SOQL.cls

```
@@ -488,8 +488,12 @@ public inherited sharing class SOQL implements Queryable {
     }
 
     public Object toValueOf(SObjectField fieldToExtract) {
-        builder.fields.clearAllFields();
-        return with(fieldToExtract).toObject()?.get(fieldToExtract);
+        try {
+            builder.fields.clearAllFields();
+            return with(fieldToExtract).toObject()?.get(fieldToExtract);
+        } catch (Exception e) {
+            return null;
```

This will suppress all errors, not only when there are no results